### PR TITLE
fix for experiment schedular not working due to cyclic services depenency and findby() old syntax

### LINKED ISSUE #2135: Experiment Schedular is not working as expected
**Version where bug was found:**
e.g "6.0.1"

**Describe the bug**
There is an issue for scheduling an experiment. The schedule jobs are not getting stored in the `scheduled_jobs` table due to the scheduledJobService being undefined in the ExperimentService due to cyclic dependency of both these services in each other. Also, the findBy() function syntax is not updated as per typeorm v0.3.

**To Reproduce**
1. Try creating a scheduled experiment.
2. The experiment would not be created

**Expected behavior**
Schedular should work.

### PR DESCRIPTION
**Pull request by ppratikcr7**
This PR resolves #2135

### CODE REVIEW

File: backend/packages/Upgrade/src/api/services/ScheduledJobService.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export class ScheduledJobService {
     return await this.dataSource.transaction(async (transactionalEntityManager) => {
       try {
         const scheduledJobRepository = transactionalEntityManager.getRepository(ScheduledJob);
-        const scheduledJob = await scheduledJobRepository.findOne({ where: { id }, relations: ['experiment'] });
+        const scheduledJob = await scheduledJobRepository.findOne({ where: { id: id }, relations: ['experiment'] });
```
[the remark this answers is not in the document]
Comment (ppratikcr7) [reply backend/packages/Upgrade/src/api/services/ScheduledJobService.ts:78]: This new syntax was required for typeorm v0.3 upgrade. Without this the findOne would not return data.

File: backend/packages/Upgrade/src/api/services/ScheduledJobService.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ export class ScheduledJobService {
...
```
[the remark this answers is not in the document]
Comment (ppratikcr7) [reply backend/packages/Upgrade/src/api/services/ScheduledJobService.ts:31]: This new syntax was required for typeorm v0.3 upgrade. Without this the findOne would not return data.

**Comment by ppratikcr7** [comment]
@VivekFitkariwala Please check this: https://www.npmjs.com/package/typeorm
Screenshot 2024-11-29 at 10 47 32 AM

File: backend/packages/Upgrade/src/api/services/ScheduledJobService.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (VivekFitkariwala) [reply backend/packages/Upgrade/src/api/services/ScheduledJobService.ts:31]: This is already in a new syntax. The only change I can see is `{id} => {id: id}` which is redundant.

[the remark this answers is not in the document]
Comment (ppratikcr7) [reply backend/packages/Upgrade/src/api/services/ScheduledJobService.ts]: @VivekFitkariwala Fixed. Thanks.

**Review by VivekFitkariwala — APPROVED** [review]

**Pull request merged**